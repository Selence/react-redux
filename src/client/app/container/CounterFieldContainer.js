import React from 'react';
import {connect} from 'react-redux'
import { CounterFieldComponent } from '../components/CounterFieldComponent'

/**
 * @param {Object} state
 * @returns {Object}
 */
const mapStateToProps = (state) => {
    return {
        counter: state.counter.value || 0
    }
};

export default connect(mapStateToProps)(CounterFieldComponent)
