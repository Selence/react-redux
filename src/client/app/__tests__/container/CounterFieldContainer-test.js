import React from 'react';
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import CounterFieldContainer from '../../container/CounterFieldContainer';

jest.mock(
    '../../components/CounterFieldComponent',
    () => ({
        CounterFieldComponent: (props) => (<div className="CounterFieldComponent" {...props} />)
    })
);

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Container CounterField', () => {
    let fakeStore = {};

    const renderComponent = () => (
        renderer.create(
            <Provider store={mockStore(fakeStore)}>
                <CounterFieldContainer />
            </Provider>
        )
    );

    describe('when the container is called', () => {
        beforeEach(() => {
            fakeStore = {
                counter: {}
            }
        });

        describe('with a counter value', () => {
            beforeEach(() => {
                fakeStore.counter.valueOf = 5;
            });

            it('the props are parsed to the component', () => {
                expect(renderComponent()).toMatchSnapshot();
            });
        });

        describe('with no counter value', () => {
            it('the default props are parsed to the component', () => {
                expect(renderComponent()).toMatchSnapshot();
            });
        });
    })
});
