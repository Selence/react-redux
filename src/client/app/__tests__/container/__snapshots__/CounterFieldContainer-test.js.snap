// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container CounterField when the container is called with a counter value the props are parsed to the component 1`] = `
<div
  className="CounterFieldComponent"
  counter={0}
  dispatch={[Function]}
/>
`;

exports[`Container CounterField when the container is called with no counter value the default props are parsed to the component 1`] = `
<div
  className="CounterFieldComponent"
  counter={0}
  dispatch={[Function]}
/>
`;
